### ISSUE TITLE
Raspbery Pi 2 Node Red Installation problems

### ISSUE THREAD

**Issue by scargill** [node-red/node-red#616]
Hi
I'm making a brand Raspberry new installation - starting with NOOB - (Raspbian), adding webmin, mosquitto - all is well (16gig card)... and then I came to install Node-Red. I installed Node-Red on my other Pi and as you know all is well there.

I followed the instructions here
http://nodered.org/docs/hardware/raspberrypi.html

After finishing - on this installation:

node-red-pi -v --max-old-space-size=128 runs - but says it cannot find 'arduino-firmata', 'redis' or 'mongodb'. 

I tried npm update and tried installing them manually - same names - looked like something was happening but when I run node-red-pi same issue (it runs otherwise). 

Here's the thing - in my first installation - node-red is sitting in /PI directory - I can't see it this time... yet if I do a winSCP search for .config.jason -( from the base of the file system) the search says it is in .home/pi/.node-red/ and yet I cannot see it in that directory - and I cannot see a home/pi/node-red directory .

Thoughts?

**Comment by knolleary** [comment]
Hi Peter, please use the mailing list, just as you have been, for these
sorts of questions.

Nick

**Comment by scargill** [comment]
Ok don’t know how to delete this one – I’ll leave that with you.

**Comment by knolleary** [comment]
(sorry, my phone battery was dying so wanted to point you back to the mailing list before it went - got to my laptop now so I can give a more complete answer).

arduino-firmata, redis and mongodb modues are not included in the default install - so those errors are currently to be expected and not an issue. If you _want_ to use the redis/mongo or arduino nodes, then you would have to npm install those modules explicitly - ie `npm install mongodb`. Running `npm  install` by itself only installs modules listed in the `package.json` file.

On linux systems, files that begin with a "." are hidden files. They do not get listed by default, so a simple listing may not see them.

(I assume you use of .config.**jason** rather than .**json** is autocorrect on whatever device you're using to write the issue).


If you are logged into the Pi, run: `ls -la /home/pi/.node-red` - this will list all files, including hidden ones.

To delete this issue, you click the 'close' button at the bottom - which I'll do now.

**Comment by scargill** [comment]
Hi

 

I have used npm to install node-red-node-dweeio… and I used the same to ensure the others were installed – ie arduino-firmata, redis and mongodb. I have these in my first installation.

 

There is no node-red directory under PI,  there is a node_modules directory with Arduino_firmata, mongodb, node-red-node-dweetio and redis directories.. yet node-red RUNS – still says the same 3 are missing – and has no knowledge of dweetio at all  (and I use dweetio in my first installation so I know it works.

 

??

**Comment by hardillb** [comment]
The node-red directory is actually .node-red (note the leading full stop). This hides it from normal directory listings. run "ls -a" in the pi directory and it should show all the hidden directories.

**Comment by scargill** [comment]
Thanks for that Ben. This must be new as a couple of weeks ago when I did my first node-red installation on another P2, it was just node-red and I’ve used it, added nodes, made my own etc… 

 

So now I know where .node-red is I reied deleting the .config.json,   I ran node-red,  it created a new .config.json – and the three items are still missing and dweetio is still no-where to be seen.

 

Any ideas?
